compat/os/mingw.h

```diff
@@ -48,5 +48,7 @@ fsync(int fd)
 
 #define mkdir(p,F) mkdir((p))
 
+void WIN32_maperror(unsigned long WIN32_oserrno);
```

Comment:
If this is a system-provided function, then please `#include` the right system header instead.

If this function is defined in compat/mswindows.cc, then how will MinGW build get that definition from behind its _SQUID_WINDOWS_ guard?

Reply:
You're right; I didn't notice because I don't even get to the linking phase for squid.
That function could also use an std::unordered_map and NoMoreGlobals implementation

Reply:
moving to own header/implementation